src/library/dao/playlistdao.cpp

```diff
@@ -863,7 +863,9 @@ void PlaylistDAO::shuffleTracks(const int playlistId, const QList<int>& position
     qsrand(seed);
     QHash<int,TrackId> trackPositionIds = allIds;
     QList<int> newPositions = positions;
+    QList<int> tableIDList;
     const int searchDistance = math_max(trackPositionIds.count() / 4, 1);
+    createTableIDList(&tableIDList, playlistId);
```

Comment:
Ids are unique for the given entry, but we need the playlistId, which IDs we need for which playlistID